src/Package.zig

```diff
@@ -503,15 +520,26 @@ fn fetchAndUnpack(
         if (gop.found_existing) {
             if (gop.value_ptr.*) |mod| {
                 gpa.free(build_root);
-                return .{
-                    .mod = mod,
-                    .found_existing = true,
-                };
+                switch (mod) {
+                    .zig_pkg => |pkg| return .{
+                        .mod = pkg,
+                        .found_existing = true,
+                    },
+                    .non_zig_pkg => return .{
+                        .mod = null,
+                        .found_existing = true,
+                    },
+                }
             }
         }
 
         root_prog_node.completeOne();
 
+        pkg_dir.access(build_zig_basename, .{}) catch return .{
```

Comment:
You need to set `gop.value_ptr.* = .non_zig_pkg` before returning here.

Reply:
Thanks